cub/cub/device/device_select.cuh

```diff
@@ -697,6 +697,154 @@ struct DeviceSelect
                                                          stream);
   }
 
+//! @rst
+  //! Uses the ``select_op`` functor applied to ``d_flag`` to selectively copy the 
+  //! corresponding items from ``d_in`` into ``d_out``.
+  //! The total number of items selected is written to ``d_num_selected_out``.
+  //! 
+  //! - The type of ``d_flags`` must conform to the requirements of the input 
```

Comment:
@gevtushenko @miscco Please check wording here. Not sure about if the first bullet point is needed.